pykafka/simpleconsumer.py

```diff
@@ -462,6 +477,15 @@ def _handle_success(parts):
                     [part for errcode, parts in parts_by_error.iteritems()
                      for part in parts])
 
+            # release all locks to allow fetching
+            for errcode, owned_partitions in parts_by_error.iteritems():
```

Comment:
That is a good point. I think this is a big enough issue to warrant its own ticket.